tests/unit/cluster/slot-migration.tcl

```diff
@@ -0,0 +1,363 @@
+proc get_open_slots {srv_idx} {
+    foreach line [split [R $srv_idx CLUSTER NODES] "\n"] {
+        set line [string trim $line]
+        if {$line eq {}} continue
+        if {[regexp {myself} $line] == 0} continue
+        set slots {}
+        regexp {\[.*} $line slots
+        return $slots
+    }
+}
+
+proc get_cluster_role {srv_idx} {
+    foreach line [split [R $srv_idx CLUSTER NODES] "\n"] {
+        set line [string trim $line]
+        if {$line eq {}} continue
+        if {[regexp {myself} $line] == 0} continue
+        set role {}
+        regexp {myself,(\w+)} $line -> role
+        return $role
+    }
+}
+
+proc wait_for_role {srv_idx role} {
+    set node_timeout [lindex [R 0 CONFIG GET cluster-node-timeout] 1]
+    # wait for a gossip cycle for states to be propagated throughout the cluster
+    after $node_timeout
```

Comment:
This (surprisingly) doesn't guarantee it will propagate since it's random, why not use cluster_config_consistent?

Reply:
TIL :)